### ISSUE TITLE
Support custom Django object managers

### ISSUE THREAD

**Issue by PeterJCLaw** [davidhalter/jedi#1588]
The new Django support unfortunately lost support for custom object managers (previously only they were offered as completions).

For example:
``` python
from django.db import models

class ThingManager(models.Manager):
    def more_things(self):
        return self.all()

class Thing(models.Model):
    name = models.TextField()

    objects = ThingManager()

Thing.objects.<tab>
```

Ideally this would offer all the usual manager completions, plus `more_things`.

Bonus points if Jedi can get this right for relations which reference `Thing` too.

**Comment by PeterJCLaw** [comment]
I've been poking this and got as far as `_create_manager_for` being the place to modify (I think). I'm not sure how best to have the code know about the `objects` assignment in the class though. I could probably pull it from the wrapped tree information, but doing that inline doesn't feel quite right. We can't just use `py__getattribute__` here because we're possibly already inside a call to that.

**Comment by davidhalter** [comment]
Yeah, good point. It might not be that easy.

> We can't just use py__getattribute__ here because we're possibly already inside a call to that.

I think what we could use is the filter information that is used in `_new_dict_filters` anyway. Something like this could do the trick:

```
filters = list(cls.get_filters(is_instance=True, include_metaclasses=False))
dct = {
    name.string_name: DjangoModelName(cls, name)
    for filter_ in reversed(list(filters))
    for name in filter_.values()
}
for f in filter:
    names = f.get('objects')
    if names:
        # The model uses custom object managers
        dcts['objects'] = names[0]  # The first name should be good enough.
        break
else:
    manager = _create_manager_for(cls)
    if manager:
        dct['objects'] = manager.name
return DictFilter(dct)
```

However that's very much just an idea, that code would need testing, experimentation and probably also a few fixes.

**Comment by davidhalter** [comment]
This was fixed #1593 and #1612.
